addons/stock/security/stock_security.xml

```diff
@@ -97,6 +97,13 @@
         <field name="domain_force">['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>
      </record>
 
+     <record model="ir.rule" id="stock_move_line_rule">
+        <field name="name">stock_move_line multi-company</field>
+        <field name="model_id" search="[('model','=','stock.move.line')]" model="ir.model"/>
+        <field name="global" eval="True"/>
+        <field name="domain_force">['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>
```

Comment:
You can remove the part ('|',('company_id','=',False)) and it will be ok, as company_id on stock.move is required.